sklearn/utils/__init__.py

```diff
@@ -1009,9 +1010,15 @@ def is_scalar_nan(x):
     >>> is_scalar_nan([np.nan])
     False
     """
-    # convert from numpy.bool_ to python bool to ensure that testing
-    # is_scalar_nan(x) is True does not fail.
-    return bool(isinstance(x, numbers.Real) and np.isnan(x))
+    # Try to use numpy `np.isnan` to check for nan. It can failed in case that
+    # `x` cannot be converted to a numpy number. Fall-back to the python
+    # `math.isnan`.
+    try:
+        # convert from numpy.bool_ to python bool to ensure that testing
+        # is_scalar_nan(x) is True does not fail.
+        return bool(isinstance(x, numbers.Real) and np.isnan(x))
```

Comment:
I think we can fully replace `np.isnan` with `math.isnan` here.

```suggestion
        return bool(isinstance(x, numbers.Real) and math.isnan(x))
```

Reply:
I had the same feeling and then we can remove the try / except construct. But maybe I am missing something.

Reply:
we can remove the bool because we already have a Python boolean. Let's see if the tests would pass